pep-0657.rst

```diff
@@ -210,11 +210,17 @@ Offset semantics
 ^^^^^^^^^^^^^^^^
 
 These offsets are propagated by the compiler from the ones stored currently in
-all AST nodes. They are 1-indexed and a value of 0 will mean that the
-information is not available. Although the AST nodes use ``int`` types to store
-these values, ``uint8_t`` types will be used for storage in the new map to
-minimize storage impact. This decision allows offsets to go from 0 to 255,
-while offsets bigger than these values will be treated as missing (value of 0).
+all AST nodes. The output of the public APIs (``co_positions`` and ``PyCode_Addr2Location``)
+that deal with these attributes use 0-indexed offsets (just like the AST nodes), but the underlying
+implementation is free to represent the actual data in whatever form they choose to be most efficient.
+The error code regarding information not available is ``None`` for the ``co_positions()`` API,
+and ``-1`` for the ``PyCode_Addr2Location`` API. The availability of the information highly depends
+on whether the offsets fall under the range, as well as the runtime flags for the interpreter
+configuration.
+
+Although the AST nodes use ``int`` types to store these values, ``uint8_t`` types will be used
+for storage in the new map to minimize storage impact. This decision allows offsets to go from
+0 to 255, while offsets bigger than these values will be treated as missing (value of 0).
```

Comment:
Please, add a sentence mentioning that this is an implementation detail for the first version and that the design allows this to be lifted in the future

Reply:
See the second next paragraph. 

> As specified previously, the underlying storage of the offsets should be
considered an implementation detail, as the public APIs to obtain this values
will return either C ``int`` types or Python ``int`` objects, which allows to
implement better compression/encoding in the future if bigger ranges would need
to be supported.  This PEP proposes to start with this simpler version and
defer improvements to future work.

Reply:
Yeah, I know, but reading this sentence is still a bit confusing because it seems that is dictating the format. We should clarify what is an implementation detail and what is a public API clearly to avoid confusion

Reply:
Any suggestion on phrasing this while not repeating that paragraph?

Reply:
Maybe something along the lines of:

> The AST nodes use ``int`` types to store these values. The current implementation, however, utilizes `uint8_t` types
> as an implementation detail to minimize storage impact. This decision allows offsets to go from
> 0 to 255, while offsets bigger than these values will be treated as missing (value of 0).

Reply:
Yeah, this is great